### PR TITLE
[Merged by Bors] - feat(Algebra/Category/ModuleCat): the category of sheaves of modules has limits

### PR DESCRIPTION
**Pull request by joelriou**
---

- [x] depends on: #12264
- [x] depends on: #11669

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/Category/ModuleCat/Sheaf/Limits.lean [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+variable [Small.{v} D]
+
+instance hasLimitsOfShape : HasLimitsOfShape D (SheafOfModules.{v} R) where
+
+noncomputable instance evaluationPreservesLimitsOfShape (X : Cᵒᵖ) :
+    PreservesLimitsOfShape D (evaluation R X : SheafOfModules.{v} R ⥤ _) where
+
+noncomputable instance forgetPreservesLimitsOfShape :
+    PreservesLimitsOfShape D (forget.{v} R) where
+
```
Comment (erdOne) [inline Mathlib/Algebra/Category/ModuleCat/Sheaf/Limits.lean:88]: Maybe also some `HasLimitsOfSize` instances?
By the way we should develop the instances `HasLimitOfSize.{v}` -> `[Small.{v} D] : HasLimitsOfShape D` somewhere, unless the typeclass inference is not good at dealing with universe parameters.

Comment (joelriou) [reply Mathlib/Algebra/Category/ModuleCat/Sheaf/Limits.lean:88]: Thanks for the suggestion. I have added the `HasLimitsOfSize` instance. However, I do not think that `HasLimitOfSize.{_, v} -> [Small.{v} D] : HasLimitsOfShape D` can be a good instance, as the universe `v` is unknown.

Comment (erdOne) [reply Mathlib/Algebra/Category/ModuleCat/Sheaf/Limits.lean:88]: Ah of course.
Also there is a `PreservesLimitsOfSize` typeclass as well that I forgot to mention.

Comment (erdOne) [reply Mathlib/Algebra/Category/ModuleCat/Sheaf/Limits.lean:88]: Or just `PreservesLimits` and `HasLimits` which are abbrevs of these with the usual universe parameters.

Comment (joelriou) [reply Mathlib/Algebra/Category/ModuleCat/Sheaf/Limits.lean:88]: I have added the `PreservesLimitsOfSize` and `HasLimitsOfSize`. (The `PreservesLimits/HasLimits` should be found automatically when the universes are ok.)

**Comment by erdOne** [comment]
Thanks!
maintainer merge

**Comment by riccardobrasca** [comment]
Thanks!

bors d+

**Comment by joelriou** [comment]
Thanks!

bors merge

**Pull request closed without merging**